src/EditorFeatures/CSharpTest/Formatting/FormattingEngineTests.cs

```diff
@@ -1633,6 +1634,23 @@ protected void Render()
             await AssertFormatAsync(expected, code, spans: null);
         }
 
+        [WpfTheory]
+        [CombinatorialData]
+        [WorkItem(59637, "https://github.com/dotnet/roslyn/issues/59637")]
+        public async Task FormatAttributeAtEndOfFile(bool trailingNewLine)
+        {
+            var endOfFile = trailingNewLine ? Environment.NewLine : "";
+            var code = $@"using System.Diagnostics.CodeAnalysis;
+
+[assembly:SuppressMessage(""Globalization"", ""CA1308: Normalize strings to uppercase"", Justification = ""My reason"", Scope = ""member"", Target = ""~M:Method"") ] {endOfFile}";
+            var expected = $@"using System.Diagnostics.CodeAnalysis;
+
+[assembly: SuppressMessage(""Globalization"", ""CA1308: Normalize strings to uppercase"", Justification = ""My reason"", Scope = ""member"", Target = ""~M:Method"")]{endOfFile}";
+
+            await AssertFormatAsync(expected, code, spans: null);
+            await AssertFormatAsync(expected, expected, spans: null);
+        }
```

Comment:
do we need a test that "file must have final newline" is respected?

Reply:
> file must have final newline

This is not respected. dotnet-format and the editor implement support for this option separately from our formatter.